### DISCUSSION TITLE
ArchR vs Signac for calculating number of fragments per sample (nFrags)

### DISCUSSION THREAD

**Discussion by tangybat** [stuart-lab/signac#1231] [Q&A]
Hi, 

I have noticed during quality control of my atac-seq data that one of my samples, sample #4, has an abnormally low number of fragments in ArchR but not in Signac. Is there a difference between how ArchR and Signac calculates the number of fragments for each sample? How does Signac calculate this value?

Signac:
`VlnPlot(object = comb, features = c('ATAC_Fragments'), group.by = 'orig.ident',pt.size = 0,ncol=2, log=T)`
<img width="950" alt="nFrags_Signac_erasedsamples" src="https://user-images.githubusercontent.com/86893865/192689833-b9291e3c-b8f0-403d-8ce2-7002ec92fab7.png">

ArchR:
```
ArrowFiles <- createArrowFiles(
  inputFiles = inputFiles,
  sampleNames = names(inputFiles),
  filterTSS = 4,
  filterFrags = 1000, 
  addTileMat = TRUE,
  addGeneScoreMat = TRUE
)
proj <- ArchRProject(
  ArrowFiles = ArrowFiles, 
  outputDirectory = "PBMC",
  copyArrows = FALSE #This is recommended so that you maintain an unaltered copy for later usage.
)
plotGroups(
  ArchRProj = proj, 
  groupBy = "Sample", 
  colorBy = "cellColData", 
  name = "log10(nFrags)",
  plotAs = "violin",
  alpha = 0.4,
  addBoxPlot = TRUE
)
```
<img width="950" alt="nFrags_ArchR" src="https://user-images.githubusercontent.com/86893865/192689838-77302b97-e983-4124-8bce-aba93a0f4759.png">

**Comment by timoast** [comment]
What code did you run in R to calculate the number of fragments and make this plot?

**Reply by timoast** [reply]
Thanks for updating with the code information, but it's still unclear where the "ATAC_Fragments" information comes from. This is not something computed by Signac, so you must be computing it yourself or adding some metadata?

**Reply by tangybat** [reply]
Sorry, the ATAC_Fragments information came from the cellranger-arc count pipeline, which I added in as metadata.
https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/output/per_barcode_metrics

Both ATAC_Fragments (Signac) and nFrags (ArchR) should represent unique nuclear fragments.

**Reply by timoast** [reply]
Ok, so here "ATAC_Fragments" is something that's calculated by Cellranger, and I don't know how ArchR calculates the nFrag value. I'd ask the ArchR developers for more information
